module_utils/network_lsr/argument_validator.py

```diff
@@ -633,6 +670,179 @@ def _validate_post(self, value, name, result):
         return result
 
 
+class ArgValidatorIPRoutingRules(ArgValidatorDict):
+    def __init__(self, name, family=None, required=False):
+        ArgValidatorDict.__init__(
+            self,
+            name,
+            required,
+            nested=[
+                # a valid priority ranges from 0 to 0xFFFFFFFF, "-1" is also allowed
+                # to reset the priority
+                ArgValidatorNum(
+                    "priority",
+                    default_value=None,
+                    required=True,
+                    val_min=-1,
+                    val_max=0xFFFFFFFF,
+                ),
+                ArgValidatorStr(
+                    "action",
+                    default_value="to-table",
+                    enum_values=["to-table", "blackhole", "prohibit", "unreachable"],
+                ),
+                ArgValidatorRange("dport", default_value=0, val_min=0, val_max=65534),
+                ArgValidatorStr(
+                    "family",
+                    default_value="ipv4",
+                    enum_values=["ipv4", "ipv6"],
+                ),
+                ArgValidatorIP(
+                    "from", family=family, default_value=None, plain_address=False
+                ),
+                ArgValidatorNum("from_len", default_value=0, val_min=0),
+                ArgValidatorNum(
+                    "fwmark", default_value=None, val_min=0, val_max=0xFFFFFFFF
+                ),
+                ArgValidatorNum(
+                    "fwmask", default_value=None, val_min=0, val_max=0xFFFFFFFF
+                ),
+                ArgValidatorStr("iif", default_value=None),
+                ArgValidatorBool("invert", default_value=None),
+                ArgValidatorNum("ipproto", default_value=None, val_min=0, val_max=255),
+                ArgValidatorStr("oif", default_value=None),
+                ArgValidatorRange("sport", default_value=0, val_min=0, val_max=65534),
```

Comment:
usually, the `default_value` should be a value that is also detected by the validator... or a special sentinel like `None`.
For example, `ArgValidatorNum` should only return numbers of the requested `numeric_type` *plus*, as special case, `None` or sometimes `ArgValidator.DEFAULT_VALUE` or whatever makes sense.

I mean, the ArgValidator implementation allows you to return any default value. It seems ugly that this `ArgValidatorRange` (currently) either returns a dictionary `{ "range_start":..., "range_end":...}` or numeric `0`.

Also, `ArgValidatorRange.__init__()` has `default_value=ArgValidator.DEFAULT`. If all uses of that constructor need to choose a *different* default value, it's probably not a good default value.